src/shared/components/ActionBar/ActionBar.tsx

```diff
@@ -1,103 +1,132 @@
-import React, { ReactNode } from 'react'
+import React from 'react'
+import { CSSTransition } from 'react-transition-group'
 
+import { useMediaMatch } from '@/hooks/useMediaMatch'
 import { SvgGlyphInfo } from '@/shared/icons'
+import { transitions } from '@/shared/theme'
 
 import {
-  DetailsContainer,
+  ActionBarContainer,
+  ActionButtonPrimary,
   DetailsIconWrapper,
-  StyledActionBarContainer,
-  StyledButtonsContainer,
-  StyledInfoContainer,
-  StyledInnerContainer,
+  DraftsBadgeContainer,
+  FlexWrapper,
   StyledPrimaryText,
   StyledSecondaryText,
 } from './ActionBar.style'
 
-import { Button } from '../Button'
+import { Button, ButtonProps } from '../Button'
 import { Text } from '../Text'
-import { Tooltip } from '../Tooltip'
+import { Tooltip, TooltipProps } from '../Tooltip'
+
+type ActionDialogButtonProps = {
+  text?: string
+  disabled?: boolean
+  onClick?: (e: React.MouseEvent) => void
+  tooltip?: TooltipProps
+} & Omit<ButtonProps, 'children'>
+
+type ActionDialogDraftBadge = {
+  text: string
+  tooltip?: TooltipProps
+  visible?: boolean
+}
 
 export type ActionBarProps = {
   primaryText?: string
   secondaryText?: string
-  primaryButtonTooltipText?: {
-    headerText: string
-    text: string
-    icon: boolean
-  }
-  primaryButtonText?: string
-  detailsText?: string
-  tooltipText?: string
-  detailsTextIcon?: ReactNode
-  isActive?: boolean
   fullWidth?: boolean
-  secondaryButtonText?: string
-  secondaryButtonIcon?: ReactNode
   className?: string
-  disabled?: boolean
-  onConfirmClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
-  onCancelClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+  primaryButton?: ActionDialogButtonProps
+  secondaryButton?: ActionDialogButtonProps & { visible?: boolean }
+  draftBadge?: ActionDialogDraftBadge
 }
 
-export const ActionBar: React.FC<ActionBarProps> = ({
-  primaryText,
-  secondaryText,
-  primaryButtonText,
-  secondaryButtonText,
-  detailsText,
-  tooltipText,
-  detailsTextIcon,
-  secondaryButtonIcon,
-  className,
-  disabled,
-  onConfirmClick,
-  onCancelClick,
-  primaryButtonTooltipText,
-}) => {
-  const buttonNode = primaryButtonText ? (
-    <Button disabled={disabled} onClick={onConfirmClick} size="large" type="submit">
-      {primaryButtonText}
-    </Button>
-  ) : null
+export const ActionBar = React.forwardRef<HTMLDivElement, ActionBarProps>(
+  ({ primaryText, secondaryText, className, primaryButton, secondaryButton, draftBadge }, ref) => {
+    const smMatch = useMediaMatch('sm')
+    const lgMatch = useMediaMatch('lg')
```

Comment:
I see you moved the media match inside the component but now that size is handled only from within this component, I guess we could just use regular media queries in CSS? Or is there something making this hard?